core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -1,13 +1,65 @@
 package com.mesosphere.usi.core
 
-import com.mesosphere.usi.core.models.PodSpec
+import akka.NotUsed
+import akka.stream.BidiShape
+import akka.stream.scaladsl.{BidiFlow, Broadcast, GraphDSL}
+import com.mesosphere.usi.models.{Mesos, SpecEvent, USIStateEvent}
 
-import scala.concurrent.Future
 
-class Scheduler {
+/*
+ * Provides the scheduler graph component. The component has two inputs, and two outputs:
+ *
+ * Input:
+ * 1) SpecEvents - Used to replicate the specification state from the framework to the USI scheduler
+ * 2) MesosEvents - Events from Mesos; offers, task status updates, etc.
+ *
+ * Output:
+ * 1) USIStateEvents - Used to replicate the state of pods, agents, and reservations to the framework
+ * 2) MesosCalls - Actions, such as revive, kill, accept offer, etc., used to realize the specification.
+ *
+ * Fully wired, the graph looks like this at a high-level view:
+ *
+ *                                                 *** SCHEDULER ***
+ *                    +------------------------------------------------------------------------+
+ *                    |                                                                        |
+ *                    |  +------------------+          +-------------+         USIStateEvents  |
+ *        SpecEvents  |  |                  |          |             |      /------------------>----> (framework)
+ * (framework) >------>-->                  |          |             |     /                   |
+ *                    |  |                  | Effects  |             |    /                    |
+ *                    |  |  SchedulerLogic  o----------> Persistence o---+                     |
+ *                    |  |                  |          |             |    \                    |
+ *       MesosEvents  |  |                  |          |             |     \   MesosCalls      |
+ *       /------------>-->                  |          |             |      \------------------>
+ *      /             |  |                  |          |             |                         |\
+ *     /              |  +------------------+          +-------------+                         | \
+ *    /               |                                                                        |  \
+ *   |                +------------------------------------------------------------------------+  |
+ *    \                                                                                           |
+ *     \                               +----------------------+                                  /
+ *      \                              |                      |                                 /
+ *       \-----------------------------<        Mesos         <---------------------------------
+ *                                     |                      |
+ *                                     +----------------------+
```

Comment:
This is really cool. Do you have the PlantUML for it?

Reply:
Thanks! No, PlantUML cannot generate graphs like this. I drew it with Emacs artist-mode